### PR TITLE
[WIP] Follow-up of #33 to implement the Page option

### PR DESCRIPTION
**Pull request by Pierstoval**
Fixes #33

### CODE REVIEW

File: src/References.php [hunk 1]
```diff
@@ -40,7 +40,12 @@ public function __construct()
...
         $config = require $referenceFile;
 
-        $keysToCheck = ['colors', 'interlace_types'];
+        $keysToCheck = [
+            'colors',
+            'colorspace_values',
+            'interlace_types',
+            'paper_sizes',
+        ];
         $keysExist = true;
```
Comment (pbories) [inline src/References.php:45]: OK, I was working on implementing the references too, but I had forgotten this...

Comment (Pierstoval) [reply src/References.php:45]: Np, sometimes, more brains are useful 😉

**Comment by VincentLanglet** [comment]
HI @Pierstoval what's missing to this PR ? :)

**Comment by Pierstoval** [comment]
Supporting more different values for the `-page` option!

For instance, it doesn't yet supports the `Letter+43+43` value, which should be valid.

And of course, more tests for that purpose 😉

**Comment by VincentLanglet** [comment]
> Supporting more different values for the `-page` option!
> 
> For instance, it doesn't yet supports the `Letter+43+43` value, which should be valid.

If I understand correctly, it can be anything like `foo+d1+d2` where `foo` is in `paper_sizes` and `d1` and `d2` are digits ?

It's what you were trying to do with
```
        $paperSizesRegex = '(?<papersize>'.\implode('|', $this->getPaperSizes()).')';
        $offsetRegex = '[-+]\d+';
        $offsetsRegex = \sprintf("(?<offsetx>%s)?(?<offsety>%s)?", $offsetRegex, $offsetRegex);
        $ratioRegex = '(?<ratio>[^!<>])?';
```
isn't it ?

**Comment by Pierstoval** [comment]
Exactly!

**Pull request merged**